src/runtime/runtime_nintendoswitch_svc.go

```diff
@@ -0,0 +1,29 @@
+// +build nintendoswitch
+
+package runtime
+
+import (
+	"device/arm"
+	"unsafe"
+)
+
+// Result nxOutputString(const char *str, u64 size)
+func NxOutputString(str string) uint64 {
+	strData := (*_string)(unsafe.Pointer(&str))
+
+	res := arm.AsmFull(
+		`mov x0, {str}
+         mov x1, {size}
+         svc 0x27`,
```

Comment:
Looking at this again, this cannot be safe. The compiler has no idea that registers `x0` and `x1` are being overwritten. I would suggest switching back to calling a special function in assembly (at least until `SVCall2` is supported on arm64, which does tell the compiler that a number of registers may be overwritten).

Reply:
Reverted to assembly code :D